code-challenges.test.js

```diff
@@ -11,41 +11,122 @@
+const expect1 = ["blue", "green", "yellow", "pink"];
+const expect2 = ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"];
+
+describe("shuffleArray", () => {
+  it("takes in an array, removes the first item from the array and shuffles the remaining content", () => {
+    expect(shuffleArray(colors1)).toEqual(expect.arrayContaining(expect1));
+    expect(shuffleArray(colors2)).toEqual(expect.arrayContaining(expect2));
+  });
+});
+
+// shuffleArray › takes in an array, removes the first item from the array and shuffles the remaining content
+// ReferenceError: shuffleArray is not defined
 
 // b) Create the function that makes the test pass.
+// Psuedo code:
+// declaring a function with one parameter(array of strings "") that will return the array shuffle minus the first index [index 0]
+const shuffleArray = (array) => {
+  // preforming a removal of the first index [index 0]
+  let newArray = array.slice(1);
+  // console.log(newArray);= ---> (show me something)
+  // using the `for loop` we will look through each index.
+  for (
+    let currentIndex = newArray.length - 1;
+    currentIndex > 0;
+    currentIndex--
+  ) {
+    // we are getting a random index that is not higher than the current index.
+    // As the function runs, the system is grabbing the random index to reassign as the current index.
+    // once it hit the last index it will return the shuffle array.
+    const randomIndex = Math.floor(Math.random() * currentIndex);
+    const tempValue = newArray[currentIndex];
+    newArray[currentIndex] = newArray[randomIndex];
+    newArray[randomIndex] = tempValue;
+  }
+  return newArray;
+};
 
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
+// Psuedo code:
+// We will create a function that takes in an object containing upVotes and downVotes
+// Two variables are currently provided
+// we will test suite with expect statements for each of the variables
+// we would need to implament a function that calculates the end tally.
+// subtract the number of downVotes from the number of upVotes
+
 // a) Create a test with expect statements for each of the variables provided.
 
-const votes1 = { upVotes: 13, downVotes: 2 }
+const votes1 = { upVotes: 13, downVotes: 2 };
 // Expected output: 11
-const votes2 = { upVotes: 2, downVotes: 33 }
+const votes2 = { upVotes: 2, downVotes: 33 };
 // Expected output: -31
 
+describe("takeVotesTally", () => {
+  it("takes in an object that contains up votes and down votes and returns the end tally.", () => {
+    expect(takeVotesTally(votes1)).toEqual(11);
+    expect(takeVotesTally(votes2)).toEqual(-31);
+  });
+});
+//  ReferenceError: takeVotesTally is not defined
+
 // b) Create the function that makes the test pass.
 
+const takeVotesTally = (votes) => {
+  let { upVotes = 0, downVotes = 0 } = votes;
```

Comment:
It looks like you are destructuring the object and reassigning both values to 0.  This may not be calculating the way we hope as a result.